### ISSUE TITLE
Assertions don't trigger on empty vectors

### ISSUE THREAD

**Issue by mb706** [mllg/checkmate#174]
I want all the following to trigger, but they currently don't.
```r
assertCharacter(integer(0))
assertLogical(integer(0))
assertSubset(integer(0), letters)
```
Maybe enabling this will break some other code somewhere so this behaviour could be parameterized I guess (or maybe allow these whenever `null.ok` is `TRUE`?).

**Comment by tdeenes** [comment]
Even more worrisome example:
```r
assertInteger(data.table::data.table()) # this passes
```

**Comment by mllg** [comment]
> ```r
> assertInteger(data.table::data.table()) # this passes
> ```

This is a clear bug and already fixed as part of #199 .

**Comment by mllg** [comment]
With #199, you can now set `typed.missing = TRUE` to trigger a type check on empty vectors.
